`gradle clean` doesn't work

Running `gradle clean` blows up with:

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':integrationTests:buildSrc:gretty-integrationTest:compileGroovy'.
> Could not resolve all files for configuration ':integrationTests:buildSrc:gretty-integrationTest:compileClasspath'.
   > Could not find org.gretty:gretty:3.0.5-SNAPSHOT.
     Searched in the following locations:
       - file:/home/boris/code/gretty/build/privateRepo/org/gretty/gretty/3.0.5-SNAPSHOT/maven-metadata.xml
       - file:/home/boris/code/gretty/build/privateRepo/org/gretty/gretty/3.0.5-SNAPSHOT/gretty-3.0.5-SNAPSHOT.pom
       - https://repo.maven.apache.org/maven2/org/gretty/gretty/3.0.5-SNAPSHOT/maven-metadata.xml
       - https://repo.maven.apache.org/maven2/org/gretty/gretty/3.0.5-SNAPSHOT/gretty-3.0.5-SNAPSHOT.pom
     Required by:
         project :integrationTests:buildSrc:gretty-integrationTest
```

I believe it's because of [this](https://github.com/gretty-gradle-plugin/gretty/blob/gretty-3.x/integrationTests/buildSrc/gretty-integrationTest/build.gradle#L9) line.

Unfortunately, not being able to do that leads to problems like [this](https://github.com/gretty-gradle-plugin/gretty/pull/210#issuecomment-826131978).

@f4lco - any ideas how to fix this?

--- a/build.gradle
+++ b/build.gradle
@@ -99,9 +99,16 @@ task testAllIntegrationTests(type: GradleBuild) {
   onlyIf { !project.tasks.testAll.getState().getFailure() }
 }
 
-task cleanIntegrationTests(type: GradleBuild) {
-  dir = file('integrationTests')
-  tasks = [ 'clean' ]
+task cleanIntegrationTests() {
+  doLast {
+    def directoriesToDelete = []
+    file('integrationTests').eachDirRecurse {
+      if (it.name == 'build') {
+        directoriesToDelete << it
+      }
+    }
+    directoriesToDelete.each { it.deleteDir() }
+  }
   project.tasks.clean.dependsOn it
 }
 

